test/mocha-junit-reporter-spec.js

```diff
@@ -120,411 +173,426 @@ describe('mocha-junit-reporter', function() {
+  it('can output to the console', function(done) {
+    var reporter = createReporter({mochaFile: 'test/output/console.xml', toConsole: true});
 
-    var stdout = testConsole.stdout.inspect();
-    try {
-      executeTestRunner();
-      verifyMochaFile(filePath);
-    } catch (e) {
-      stdout.restore();
-      throw e;
-    }
+    var stdout = mockStdout();
+    executeTestRunner(reporter.runner, function() {
+      verifyMochaFile(reporter.runner, filePath);
 
-    stdout.restore();
+      var xml = stdout.output[0];
+      assertXmlEquals(xml, mockXml(reporter.runner.stats));
 
-    var xml = stdout.output[0];
-    expect(xml).xml.to.be.valid();
-    expect(xml).xml.to.equal(mockXml(runner.stats));
+      done();
+    });
   });
 
-  it('properly outputs tests when amount of tests is wrong', function() {
-    createReporter({mochaFile: 'test/mocha.xml'});
-    // emulates exception in before each hook
-    executeTestRunner({skipPassedTests: true});
-
-    verifyMochaFile(filePath, {skipPassedTests: true});
+  it('properly outputs tests when error in beforeAll', function(done) {
+    var reporter = createReporter();
+    var rootSuite = reporter.runner.suite;
+    var suite1 = Suite.create(rootSuite, 'failing beforeAll');
+    suite1.beforeAll('failing hook', function() {
+      throw new Error('error in before');
+    });
+    suite1.addTest(createTest('test 1'));
+
+    var suite2 = Suite.create(rootSuite, 'good suite');
+    suite2.addTest(createTest('test 2'));
+
+    reporter.runner.run(function() {
+      expect(reporter._testsuites).to.have.lengthOf(3);
+      expect(reporter._testsuites[1].testsuite[0]._attr.name).to.equal('failing beforeAll');
+      expect(reporter._testsuites[1].testsuite[1].testcase).to.have.lengthOf(2);
+      expect(reporter._testsuites[1].testsuite[1].testcase[0]._attr.name).to.equal('failing beforeAll "before all" hook: failing hook');
+      expect(reporter._testsuites[1].testsuite[1].testcase[1].failure._attr.message).to.equal('error in before');
+      expect(reporter._testsuites[2].testsuite[0]._attr.name).to.equal('good suite');
+      expect(reporter._testsuites[2].testsuite[1].testcase).to.have.lengthOf(1);
+      expect(reporter._testsuites[2].testsuite[1].testcase[0]._attr.name).to.equal('good suite test 2');
+      done();
+    });
   });
 
   describe('when "useFullSuiteTitle" option is specified', function() {
-    var suiteTitles = ['test suite', 'when has parent'];
-
-    it('generates full suite title', function() {
-      var reporter = configureReporter({useFullSuiteTitle: true });
-
-      expect(suiteName(reporter.suites[0])).to.equal(suiteTitles[0]);
-      expect(suiteName(reporter.suites[1])).to.equal(suiteTitles.join(' '));
+    it('generates full suite title', function(done) {
+      var reporter = createReporter({useFullSuiteTitle: true });
+      executeTestRunner(reporter.runner, function() {
+        expect(suiteName(reporter._testsuites[0])).to.equal('');
```

Comment:
This may be a bug. The root suite is output with name `''` but child suites use the `Root Suite` prefix. This wasn't caught before since the stubs typically didn't have a root suite, even though Mocha always has one (AFAIK).